packages/app-builder/src/repositories/TestRunRepository.ts

```diff
@@ -0,0 +1,206 @@
+import { MarbleCoreApi } from '@app-builder/infra/marblecore-api';
+import {
+  adaptTestRun,
+  adaptTestRunCreateInputDto,
+  adaptTestRunDecision,
+  adaptTestRunRuleExecution,
+  TestRunDecision,
+  TestRunRuleExecution,
+  TestRunStatus,
+  testRunStatuses,
+  type TestRun,
+  type TestRunCreateInput,
+} from '@app-builder/models/testrun';
+import { toUUID } from '@app-builder/utils/short-uuid';
+import { addDays } from 'date-fns';
+import { randomInteger } from 'remeda';
+import short from 'short-uuid';
+
+export interface TestRunRepository {
+  getTestRun(args: { testRunId: string }): Promise<TestRun>;
+  launchTestRun(args: TestRunCreateInput): Promise<TestRun>;
+  listTestRuns(args: { scenarioId: string }): Promise<TestRun[]>;
+  listDecisions(args: { testRunId: string }): Promise<TestRunDecision[]>;
+  listRuleExecutions(args: {
+    testRunId: string;
+  }): Promise<TestRunRuleExecution[]>;
+}
+
+const testruns: TestRun[] = [
+  {
+    id: '989ed5e4-c7ca-4685-9535-a3e1ab9dfc75',
+    refIterationId: '6f6fe0d8-9a1a-4d5a-bdd7-fa7fcda1b4e3',
+    scenarioId: '6f6fe0d8-0804-4500-ae68-f4e56ea780d7',
+    testIterationId: '6f6fe0d8-bbc8-4df3-a913-c0064ed99e4e',
+    startDate: String(new Date().getTime()),
+    endDate: String(addDays(new Date(), 1).getTime()),
+    creatorId: '96762987-8895-4af2-9c0a-2dffde09985c',
+    status: 'up',
+  },
+];
+
+const testrunDecisions: TestRunDecision[] = [
+  { version: 'V1', outcome: 'approve', count: 10 },
+  { version: 'V4', outcome: 'approve', count: 20 },
+  { version: 'V1', outcome: 'decline', count: 5 },
+  { version: 'V1', outcome: 'approve', count: 30 },
+  { version: 'V4', outcome: 'decline', count: 15 },
+  { version: 'V1', outcome: 'review', count: 9 },
+  { version: 'V4', outcome: 'review', count: 22 },
+  { version: 'V1', outcome: 'block_and_review', count: 20 },
+];
+
+const testrunRuleExecutions: TestRunRuleExecution[] = [
+  {
+    version: 'V1',
+    name: 'Rule 1 name',
+    status: 'hit',
+    total: 10,
+    rule_id: 'rule-1',
+  },
+  {
+    version: 'V4',
+    name: 'Rule 1 name',
+    status: 'hit',
+    total: 15,
+    rule_id: 'rule-1',
+  },
+  {
+    version: 'V1',
+    name: 'Rule 1 name',
+    status: 'no_hit',
+    total: 20,
+    rule_id: 'rule-1',
+  },
+  {
+    version: 'V4',
+    name: 'Rule 1 name',
+    status: 'no_hit',
+    total: 15,
+    rule_id: 'rule-1',
+  },
+  {
+    version: 'V1',
+    name: 'Rule 1 name',
+    status: 'error',
+    total: 5,
+    rule_id: 'rule-1',
+  },
+  {
+    version: 'V4',
+    name: 'Rule 1 name',
+    status: 'snoozed',
+    total: 5,
+    rule_id: 'rule-1',
+  },
+  {
+    version: 'V1',
+    name: 'Rule 2 name',
+    status: 'no_hit',
+    total: 5,
+    rule_id: 'rule-2',
+  },
+  {
+    version: 'V4',
+    name: 'New Rule 2 name',
+    status: 'no_hit',
+    total: 0, // I don't know if this is a possible return from the backend but I'm adding it here to test the UI
```

Comment:
Good remark - the backend in fact should not return "0" rows, so the front should work in that context too

Reply:
It does works ;)